Simplenote/StorageError.swift

```diff
@@ -0,0 +1,35 @@
+import Foundation
+
+enum StorageError: Error {
+    case appConfigurationError
+    case attachingPersistentStoreFailure
+    case fetchError
+}
+
+extension StorageError {
```

Comment:
Nitpciky: Most of these errors aren't really Storage Errors:

- `appConfigurationError` is thrown when the app is logged off
- `fetchError` is thrown by WidgetDataController if fetch fails

I think we could prob. get rid of this enum. Where is it actually used? (still reading!!)

Reply:
Currently this gets called when the core data manager fails to load the persistent store, if the widget data controller fails to fetch, and if the app is not logged in.

So here is my proposal for these error comments.  First if I remove the persistentStore error case from this all of these errors are specific to the widgets.  So I could change the name of the error to WidgetError and still use the configuration and fetch errors in the places they currently are.
Then in the core data manager I can throw `throw NSError(domain: Constants.errorDomain, code: .zero, userInfo: nil)` instead

I have updated the codes for that.  WDYT?

Reply:
Swift Errors are absolutely great. Specifically, the one thing that caught my eye is  the name. It's called `StorageError`, but it may be thrown by WidgetDataController.

StorageError would maybe imply it came from the Storage* (StorageManager perhaps?)